src/connection.rs

```diff
@@ -42,6 +42,12 @@ impl Connection {
             }
         }
         let stream = TcpStream::connect(addr.clone())?;
+        let tcp_nodelay = addr.query_pairs()
+            .find(|&(ref k, ref v)| {
+                k == &Cow::Borrowed("tcp_nodelay") && v == &Cow::Borrowed("true")
```

Comment:
We can compare them by `k == "tcp_nodelay"` since `Cow` have implemented the `Eq` trait.

Reply:
Thanks, fixed it.